### DISCUSSION TITLE
calling dialog.open inside the function doesn't work

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#492] [Q&A] [answered]
why does calling dialog.open not work when called inside function .or is there another way to write code for this

```py
with ui.dialog() as dialog , ui.card():
    jj = ui.input(label="dasds")
    ui.button("save")

def openthedialog():
    dialog.open


ui.button("open dialog",on_click=openthedialog)
```

**Comment by rodja** [comment]
You need to execute the function -> place brackets behind `dialog.open`. The `on_click` handler expects a function, thats why you are not execution it but passing a reference; otherwise you pass a lambda statement which itself is a function:

```py
ui.button("open dialog",on_click=dialog.open) # this will open the dialog
ui.button("open dialog",on_click=lambda: dialog.open()) # this will also work
```

As you can see you can also remove the opendialog function completely If you do not want to do additional things on button click.

**Comment by bobwatcherx** [comment, marked as answer]
yes it works thanks
code
```
from nicegui import ui

with ui.dialog() as dialog , ui.card():
    jj = ui.input(label="dasds")
    ui.button("save")

def openthedialog():
    dialog.open()
    print("123123")


ui.button("open dialog",on_click=openthedialog)


ui.run()
```
